### PR TITLE
Insta Simulation Dex Connector

### PR DESCRIPTION
**Pull request by pradyuman-verma**
- [x] Review
- [x] Deploy
- [x] Add to SDK https://github.com/Instadapp/dsa-connect/pull/194
- [x] Enable

### CODE REVIEW

File: contracts/polygon/connectors/dexSimulation/main.sol [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		uint256 getId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		sellAmount = getUint(getId, buyAmount);
+
+		if (sellToken == maticAddr) {
+			sellAmount = sellAmount == uint256(-1)
```
Comment (thrilok209) [inline contracts/polygon/connectors/dexSimulation/main.sol:39]:
```suggestion
		uint256 nativeAmount
		if (sellToken == maticAddr) {
			sellAmount = sellAmount == uint256(-1)
```

File: contracts/polygon/connectors/dexSimulation/main.sol [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				? address(this).balance
+				: sellAmount;
+		} else {
```
Comment (thrilok209) [inline contracts/polygon/connectors/dexSimulation/main.sol:43]:
```suggestion
			nativeAmount = sellAmount;
		} else {
```

File: contracts/polygon/connectors/dexSimulation/main.sol [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			TokenInterface tokenContract = TokenInterface(sellToken);
+
+			sellAmount = sellAmount == uint256(-1)
+				? tokenContract.balanceOf(address(this))
+				: sellAmount;
+
+			approve(tokenContract, address(dexSimulation), sellAmount);
+		}
+
+		InstaDexSimulation(dexSimulation).swap{ value: sellAmount }(
```
Comment (thrilok209) [inline contracts/polygon/connectors/dexSimulation/main.sol]:
```suggestion
		InstaDexSimulation(dexSimulation).swap{ value: nativeAmount }(
```

File: contracts/polygon/connectors/dexSimulation/main.sol [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+		setUint(setId, buyAmount);
+
+		_eventName = "LogSimulateSwap(address,address,uint256,uint256)";
+		_eventParam = abi.encode(sellToken, buyToken, sellAmount, buyAmount);
+	}
+}
+
+contract ConnectV2InstaDexSimulation is InstaDexSimulationResolver {
+	string public name = "InstaDexSimulation-v1";
```
Comment (thrilok209) [inline contracts/polygon/connectors/dexSimulation/main.sol]:
```suggestion
	string public name = "Instadapp-DEX-Simulation-v1";
```

File: contracts/polygon/connectors/dexSimulation/main.sol [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import "./helpers.sol";
+
+abstract contract InstaDexSimulationResolver is Events, Helpers {
+	/**
+	 * @dev Simulation swap using Insta dex swap contract
+	 * @param sellToken The token to sell/swap
+	 * @param buyToken The token to buy
+	 * @param sellAmount The sell token amount
+	 * @param buyAmount The buy token amount
+	 * @param setId Set token amount at this ID in `InstaMemory` Contract.
```
Comment (thrilok209) [inline contracts/polygon/connectors/dexSimulation/main.sol:20]: @pradyuman-verma missing getId

**Comment by pradyuman-verma** [comment]
### Deployed on Polygon

Name: `INSTA-DEX-SIMULATION-A`
Addr: [0xf6D09B45eB7baeE70079c80C73c02F7a0f298885](https://polygonscan.com/address/0xf6D09B45eB7baeE70079c80C73c02F7a0f298885)
tx: [here](https://polygonscan.com/tx/0x637a4699f7fc02d397af390e9f923cb771f39287dfb3dbe0f8cc046b248fe9cf)

### Deployed On Arb:

Name: `INSTA-DEX-SIMULATION-A`
Addr: [0xe11d191AF6c834ce5b11a0Ea52cBc16d0fd52F1e](https://arbiscan.io/address/0xe11d191AF6c834ce5b11a0Ea52cBc16d0fd52F1e#code)
tx: [here](https://arbiscan.io/tx/0xccabf799a876c886c0acf8ceceebf66bb790156445b0d794845537171f87f784#eventlog)

### Deployed On Opt:

Name: `INSTA-DEX-SIMULATION-A`
Addr: [0x9341dd1191837C3ab30447ADa907F7625a61CF14](https://optimistic.etherscan.io/address/0x9341dd1191837C3ab30447ADa907F7625a61CF14#code)
tx: [here](https://optimistic.etherscan.io/tx/0x4a7c4602014fd2816122f294a4ef6cef3cb3f19fccc6964c42c489a60889896f#eventlog)

### Deployed On Avax:

Name: `INSTA-DEX-SIMULATION-A`
Addr: [0xCFad6dcE4335B0f125B23e3a870e2fDE05f9A90d](https://snowtrace.io/address/0xCFad6dcE4335B0f125B23e3a870e2fDE05f9A90d#code)
tx: [here](https://snowtrace.io/tx/0xa3e993597a457181b9e1ec82b10fd447aecdd69a421cd482744a34ac9109a42e)

### Deployed On FTM:

Name: `INSTA-DEX-SIMULATION-A`
Addr: [0x3595d71b4d96a155Ef08f0B7643384751AdeD9f4](https://ftmscan.com/address/0x3595d71b4d96a155Ef08f0B7643384751AdeD9f4#code)
tx: [here]()

### Deployed on Mainnet
Name: `INSTA-DEX-SIMULATION-A`
Addr: [0x6FA88564Ed767b38D029c91A56F62e80886A526c](https://etherscan.io/address/0x6FA88564Ed767b38D029c91A56F62e80886A526c)
tx: [here](https://etherscan.io/tx/0x4ceae42756a4fa2856d9310212948998773d4f6c7433f11c132458ee4b136beb)

File: contracts/polygon/connectors/dexSimulation/main.sol [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		uint256 sellAmount,
+		uint256 buyAmount,
+		uint256 setId,
+		uint256 getId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		sellAmount = getUint(getId, buyAmount);
```
Comment (pradyuman-verma) [inline contracts/polygon/connectors/dexSimulation/main.sol]:
```suggestion
		sellAmount = getUint(getId, sellAmount);
```

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**